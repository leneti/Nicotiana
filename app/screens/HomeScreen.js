import * as React from "react";
import { Image, StyleSheet, View, Text } from "react-native";
import colors from "../config/colours";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";
import AwesomeButton from "@umangmaurya/react-native-really-awesome-button";
import { Container } from "native-base";

export default function HomeScreen({ navigation }) {
  return (
    <Container>
      <View style={styles.background}>
        <View
          style={{
            alignItems: "center",
          }}
        >
          <Text style={{ fontSize: 18, color: colors.white }}>Welcome to</Text>
          <Text
            style={{
              fontSize: 45,
              color: colors.pink,
              fontFamily: "Yantramanav_300Light",
            }}
          >
            NICOTIANA
          </Text>
          <AwesomeButton
            style={{ marginTop: hp(5) }}
            onPress={() => navigation.navigate("SignIn")}
            width={wp(64)}
            borderRadius={20}
            height={45}
            textColor={colors.background}
            textSize={16}
            backgroundColor={colors.white}
            backgroundDarker={colors.grey}
            raiseLevel={3}
          >
            Log in
          </AwesomeButton>
          <AwesomeButton
            style={{ marginTop: hp(1) }}
            onPress={() => navigation.navigate("SignUp")}
            width={wp(64)}
            borderRadius={20}
            height={45}
            textColor={colors.white}
            textSize={16}
            backgroundColor={colors.pink}
            backgroundDarker={colors.pink_darker}
            raiseLevel={3}
          >
            Sign Up
          </AwesomeButton>
        </View>
        <View
          pointerEvents="none"
          style={{
            alignSelf: "flex-end",
            zIndex: -1,
            bottom: 0,
            position: "absolute",
          }}
        >
          <Image
            resizeMode="contain"
            style={styles.hookah}
            source={require("../assets/hookah.png")}
          />
        </View>
      </View>
    </Container>
  );
}

const styles = StyleSheet.create({
  background: {
    flex: 1,
    backgroundColor: colors.background,
    justifyContent: "center",
    alignItems: "center",
  },
  hookah: {
    width: 200,
    height: 366,
    alignSelf: "flex-end",
  },
});
