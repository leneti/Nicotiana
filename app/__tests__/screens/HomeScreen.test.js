import React from "react";
import renderer from "react-test-renderer";

import HomeScreen from "../../screens/HomeScreen";

describe("<HomeScreen />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
