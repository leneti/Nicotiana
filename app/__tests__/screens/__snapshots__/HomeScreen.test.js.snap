// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#36383F",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#f1f1f1",
            "fontSize": 18,
          }
        }
      >
        Welcome to
      </Text>
      <Text
        style={
          Object {
            "color": "#ff9baf",
            "fontFamily": "Yantramanav_300Light",
            "fontSize": 45,
          }
        }
      >
        NICOTIANA
      </Text>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "height": 45,
            "marginTop": 66.5,
            "opacity": 1,
            "width": 480,
            "zIndex": 10,
          }
        }
        testID="aws-btn-content-view"
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.15)",
              "borderBottomLeftRadius": null,
              "borderBottomRightRadius": null,
              "borderRadius": 20,
              "borderTopLeftRadius": null,
              "borderTopRightRadius": null,
              "bottom": -1.5,
              "height": 42,
              "left": "1%",
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
              "width": "98%",
            }
          }
          testID="aws-btn-shadow"
        />
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
              },
              Object {
                "backgroundColor": "#acacad",
                "borderBottomLeftRadius": null,
                "borderBottomRightRadius": null,
                "borderRadius": 20,
                "borderTopLeftRadius": null,
                "borderTopRightRadius": null,
                "height": 42,
                "width": 480,
              },
            ]
          }
          testID="aws-btn-bottom"
        />
        <View
          style={
            Object {
              "borderBottomLeftRadius": null,
              "borderBottomRightRadius": null,
              "borderRadius": 20,
              "borderTopLeftRadius": null,
              "borderTopRightRadius": null,
              "height": 42,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
              "width": 480,
            }
          }
          testID="aws-btn-content"
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "zIndex": 3,
                },
                Object {
                  "backgroundColor": "#f1f1f1",
                  "borderBottomLeftRadius": null,
                  "borderBottomRightRadius": null,
                  "borderColor": null,
                  "borderRadius": 20,
                  "borderTopLeftRadius": null,
                  "borderTopRightRadius": null,
                  "borderWidth": 0,
                },
              ]
            }
            testID="aws-btn-text"
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.15)",
                  "height": 42,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 480,
                }
              }
              testID="aws-btn-active-background"
            />
            <Text
              style={
                Object {
                  "color": "#36383F",
                  "fontFamily": null,
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "opacity": 1,
                  "paddingBottom": 0,
                  "paddingHorizontal": 12,
                  "paddingTop": 0,
                  "textAlign": "center",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": "100%",
                  "zIndex": 10,
                }
              }
              testID="aws-btn-content-text"
            >
              Log in
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "height": 45,
            "marginTop": 13.5,
            "opacity": 1,
            "width": 480,
            "zIndex": 10,
          }
        }
        testID="aws-btn-content-view"
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.15)",
              "borderBottomLeftRadius": null,
              "borderBottomRightRadius": null,
              "borderRadius": 20,
              "borderTopLeftRadius": null,
              "borderTopRightRadius": null,
              "bottom": -1.5,
              "height": 42,
              "left": "1%",
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
              "width": "98%",
            }
          }
          testID="aws-btn-shadow"
        />
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
              },
              Object {
                "backgroundColor": "#6e4555",
                "borderBottomLeftRadius": null,
                "borderBottomRightRadius": null,
                "borderRadius": 20,
                "borderTopLeftRadius": null,
                "borderTopRightRadius": null,
                "height": 42,
                "width": 480,
              },
            ]
          }
          testID="aws-btn-bottom"
        />
        <View
          style={
            Object {
              "borderBottomLeftRadius": null,
              "borderBottomRightRadius": null,
              "borderRadius": 20,
              "borderTopLeftRadius": null,
              "borderTopRightRadius": null,
              "height": 42,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
              "width": 480,
            }
          }
          testID="aws-btn-content"
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "overflow": "hidden",
                  "zIndex": 3,
                },
                Object {
                  "backgroundColor": "#ff9baf",
                  "borderBottomLeftRadius": null,
                  "borderBottomRightRadius": null,
                  "borderColor": null,
                  "borderRadius": 20,
                  "borderTopLeftRadius": null,
                  "borderTopRightRadius": null,
                  "borderWidth": 0,
                },
              ]
            }
            testID="aws-btn-text"
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.15)",
                  "height": 42,
                  "left": 0,
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 480,
                }
              }
              testID="aws-btn-active-background"
            />
            <Text
              style={
                Object {
                  "color": "#f1f1f1",
                  "fontFamily": null,
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "opacity": 1,
                  "paddingBottom": 0,
                  "paddingHorizontal": 12,
                  "paddingTop": 0,
                  "textAlign": "center",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": "100%",
                  "zIndex": 10,
                }
              }
              testID="aws-btn-content-text"
            >
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "alignSelf": "flex-end",
          "bottom": 0,
          "position": "absolute",
          "zIndex": -1,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "alignSelf": "flex-end",
            "height": 366,
            "width": 200,
          }
        }
      />
    </View>
  </View>
</View>
`;
